### ISSUE TITLE
Loss not changing for XGB regressor with cross validation

### ISSUE THREAD

**Issue by roastedduck73** [hyperopt/hyperopt#878]
I'm trying to build a regression model using XGBoost with Hyperopt. Based on my readings, below is how I implemented Hyperopt with cross validation. However, the loss is constant all the time :/

Attaching the code:

```
def objective(space):
 
#     print("Training with params: ")
#     print(space)
 
    cv = KFold(n_splits=5, shuffle=True, random_state = 12)
    cv_scores_rmse = np.empty(5)
    cv_scores_r2 = np.empty(5)
 
    for idx, (train_idx, test_idx) in enumerate(cv.split(x_train, y_train)):
        x_train_hat, x_test_hat = x_train.iloc[train_idx], x_train.iloc[test_idx]
        y_train_hat, y_test_hat = y_train.iloc[train_idx], y_train.iloc[test_idx]
 
        model = xgb.XGBRegressor(n_estimator = int(space['n_estimators']),
                                 max_depth = int(space['max_depth']),
                                 learning_rate = space['learning_rate'],
                                 min_child_weight = int(space['min_child_weight']),
                                 gamma = space['gamma'],
                                 reg_alpha = space['reg_alpha'],
                                 reg_lambda = space['reg_lambda'],
                                 colsample_bytree = space['colsample_bytree'],
                                 subsample = space['subsample'],
                                 verbosity = 0,
                                 seed = 12)
 
        evaluation = [(x_train_hat, y_train_hat), (x_test_hat, y_test_hat)]
 
        model.fit(x_train_hat,
                  y_train_hat,
                  eval_set=evaluation,
                  eval_metric="rmse",
                  early_stopping_rounds=10,
                  verbose = False)

        y_pred = model.predict(x_test_hat)
        cv_scores_rmse[idx] = np.sqrt(mean_squared_error(y_test_hat, y_pred))
        cv_scores_r2[idx] = r2_score(y_test_hat,y_pred)
 
    print({'val_rmse': np.mean(cv_scores_rmse), 'val_r2': np.mean(cv_scores_r2)})
 
    return {'loss': -np.mean(cv_scores_r2), 'status':STATUS_OK}
 
def _get_best_param():
    
    space ={'booster': 'gbtree',
            'n_estimators': hp.quniform('n_estimators', 10, 100, 5),
            'max_depth': hp.quniform('max_depth', 3, 13, 1),
            'learning_rate': hp.uniform('learning_rate', 0.01, 0.3),
            'min_child_weight' : hp.quniform('min_child_weight', 0, 10, 1),
            'gamma': hp.uniform ('gamma', 1, 9),
            'reg_alpha' : hp.uniform('reg_alpha', 40, 180),
            'reg_lambda' : hp.uniform('reg_lambda', 0, 1),
            'colsample_bytree' : hp.uniform('colsample_bytree', 0.5, 1),
            'subsample': hp.uniform('subsample', 0.5, 1),
            'seed':12}
 
    trials = Trials()
    best_hyperparams = fmin(fn = objective,
                                              space = space,
                                              algo = tpe.suggest,
                                              max_evals = 15,
                                              trials = trials)
    
    return best_hyperparams
```

Things I have tried:
1. printed the params set passing to the regressor, confirmed that each set of hyperparamters did change.
2. tried initiating trials = Trials() in the cross validation loop based on this post https://github.com/hyperopt/hyperopt/issues/497, didn't work.
3. tried restructuring the code using cross_val_score function based on this post https://www.kaggle.com/code/ilialar/hyperparameters-tunning-with-hyperopt/notebook, gave the same constant loss.

If anyone knows how to resolve this issue, any help here will be greatly appreciated! Thank you.
